### PR TITLE
Use the first audio source if not set by the user and remove `None` from the preferences

### LINKED ISSUE #276: Preferences for recording audio displaying "None" device
If you have not overwritten the system settings for audio recording in Kap, the selector should display "Default".

### PR DESCRIPTION
**Pull request by sallar**
Fixes #276

image

### CODE REVIEW

**Comment by sindresorhus** [comment]
There still needs to be a `None` option for people that don't want to record audio.

**Comment by sindresorhus** [comment]
And maybe `Default` could show the first part of the default device name (as much as fits including `…`). `Default` => `Default (Built-in Microphone)`

**Comment by sindresorhus** [comment]
@skllcrn ⏫ Thoughts?

**Comment by sallar** [comment]
@sindresorhus this is how it looks like now:

image

can you please review again? :)

**Comment by sallar** [comment]
@sindresorhus we've gotta think about using some other solution for dropdowns in the rewrite. native html dropdowns are not flexible. Can't show ... in the preview and full one when opened.

**Comment by sindresorhus** [comment]
> we've gotta think about using some other solution for dropdowns in the rewrite. native html dropdowns are not flexible. Can't show ... in the preview and full one when opened.

Yeah, that's not a good UX.

**Comment by sindresorhus** [comment]
screen shot 2018-01-20 at 22 04 43

There should only be one entry here. In short, if the user has never chosen an audio device, it should just be the first one.

I think we should drop the `Default` prefix and the shortening, seeing how it's impossible to show different content in the preview from the actual item.

**Comment by sindresorhus** [comment]
So in my case, the only item in the list should be "Built-in Microphone" and it should be the default when I haven't selected one yet since it's the first in the list.

**Comment by sallar** [comment]
@sindresorhus with the "default" option removed, there is a potential issue.

* If we set the `defaultAudioInputId` on app init to the first input, and user changes their default later, our selector will still show the old value. 
* If we don't set that and set it to `default` (in settings), the dropdown will not be the representation of the actual value stored, because it will change every time the user changes their setting from their system.

What do you think?

**Comment by sindresorhus** [comment]
Let's just drop the concept of default in the UI. On startup, we'll take the first input if not set.


First launch:

```
| Built-in microphone |
| External input      |
```

User selects "External input.

Second launch:

```
| External input      |
| Built-in microphone |
```

We don't have to show the word "Default" anywhere. We'll just default to the first one internally if the user has not specifically set any device.

The user changing the default system input while Kap is launched sounds like an edge-case, but we could always just use the live value directly when recording starts if no preference is set.

**Comment by sallar** [comment]
@sindresorhus Done. Now it works like you described.

File: app/src/renderer/js/main.js [hunk 1]
```diff
@@ -9,8 +10,7 @@ import {handleKeyDown, validateNumericInput} from '../js/input-utils';
 import {handleTrafficLightsClicks, isVisible, disposeObservers} from '../js/utils';
 import buildSizeMenu, {findRatioForSize} from '../js/size-selector';
 
-const aperture = require('aperture')();
-
+const aperture = createAperture();
```
Comment (sindresorhus) [inline app/src/renderer/js/main.js]:
```js
const aperture = new Aperture();
```

File: app/src/renderer/js/main.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import {ipcRenderer, remote} from 'electron';
 import EventEmitter from 'events';
+import {default as createAperture, audioDevices} from 'aperture';
```
Comment (sindresorhus) [inline app/src/renderer/js/main.js]:
```js
import Aperture from 'aperture';
```

And then just use `Aperture.audioDevices`.

Comment (sallar) [reply app/src/renderer/js/main.js]: @sindresorhus default export is `createAperture` and `Aperture` is not exported. should I send a PR to that package?

Comment (sindresorhus) [reply app/src/renderer/js/main.js]: Oh, that's true. Duuh. I can fix the docs there. Never mind then.

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**